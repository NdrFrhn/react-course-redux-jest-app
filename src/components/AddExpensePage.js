import React from 'react'
import { connect } from 'react-redux';
import { addExpense } from '../actions/expenses';
import { useNavigate } from 'react-router-dom';
import ExpenseForm from "./ExpenseForm"; 
 
export class AddExpensePage extends React.Component {
  onSubmit = (expense) => {
      this.props.addExpense(expense)
      this.props.navigate('/');
  }

  render() {
      return (
          <div>
            <h1>Add Expenses</h1>
            <ExpenseForm
              onSubmit = {this.onSubmit}
            /> 
          </div>
      );
  }
}
 
const mapDispatchToProps = (dispatch) => ({
    addExpense: (expense) => dispatch(addExpense(expense))
});
 
export function WithNavigate(props) {
    let navigate = useNavigate();
    return <AddExpensePage {...props} navigate={navigate} />
}
 
export default connect(undefined, mapDispatchToProps)(WithNavigate);