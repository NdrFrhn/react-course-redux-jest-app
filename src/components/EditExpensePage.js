import React from "react";
import { connect } from "react-redux";
import { useParams, useNavigate } from "react-router-dom";
import ExpenseForm from "./ExpenseForm";
import { editExpense, removeExpense } from "../actions/expenses";

export class EditExpensePage extends React.Component {
  onSubmit = (expense) => {
    this.props.editExpense(this.props.expense.id, expense);
    this.props.navigate('/');
  };
  onRemove = () => {
    this.props.removeExpense({id: this.props.expense.id});
    this.props.navigate('/');
  }

  render () {
    return (
      <div>
        <ExpenseForm 
          expense={this.props.expense}
          onSubmit= {this.onSubmit}
        />
        <button onClick={this.onRemove} >
          Remove
        </button>
      </div>
    )
  }
}

const mapDispatchToProps = (dispatch) => ({
  editExpense: (id, expense) => dispatch(editExpense(id, expense)),
  removeExpense: expense => dispatch(removeExpense(expense))
});

const mapStateToProps = (state, props) => ({
  expense: state.expenses.find( (expense) => expense.id === props.id )
})

export function WithNavigate (props) {
  let navigate = useNavigate();
  return <EditExpensePage {...props} navigate={navigate} />
}

export default connect(mapStateToProps, mapDispatchToProps)(WithNavigate);