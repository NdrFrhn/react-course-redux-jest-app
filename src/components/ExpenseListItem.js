import React from "react";
import { Link } from "react-router-dom";
import moment from "moment";
import numeral from "numeral";

let pound = '£';

const ExpenseListItem = ({id, description, amount, createdAt}) => (
    <div>
    <Link to={`/edit/${id }`}>
      <h3>{description} { id.substr(id.length - 4) }</h3>
    </Link>
      <p>
        {pound} { numeral(amount / 100).format('£0,0.00')} 
        - 
        {moment(createdAt).format('Do MMMM, YYYY')}
      </p>
    </div>
  );

export default ExpenseListItem;