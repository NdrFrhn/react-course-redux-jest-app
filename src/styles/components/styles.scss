@use '../base/base.scss';
@use '../base/settings';
