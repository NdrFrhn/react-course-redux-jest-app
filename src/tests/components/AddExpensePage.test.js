import React from "react";
import { shallow } from 'enzyme';
import { AddExpensePage } from "../../components/AddExpensePage";
import expenses from "../fixtures/expenses";

let addExpense, wrapper, navigate;
beforeEach(() => {
  addExpense = jest.fn();
  navigate = jest.fn();
  wrapper = shallow(<AddExpensePage navigate={navigate} addExpense={addExpense} />);
});

test('should render AddExpensePage correctly', () => {
  expect(wrapper).toMatchSnapshot();
});

test('should return the WithNavigate function correctly', () => {
  expect(wrapper).toMatchSnapshot();
});

test('should handle onSubmit', () => {
  wrapper.find('ExpenseForm').prop('onSubmit')(expenses[1]);
  expect(addExpense).toHaveBeenLastCalledWith(expenses[1]);
});