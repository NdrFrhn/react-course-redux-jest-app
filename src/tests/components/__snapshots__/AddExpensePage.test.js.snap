// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render AddExpensePage correctly 1`] = `
<div>
  <h1>
    Add Expenses
  </h1>
  <ExpenseForm
    onSubmit={[Function]}
  />
</div>
`;

exports[`should return the WithNavigate function correctly 1`] = `
<div>
  <h1>
    Add Expenses
  </h1>
  <ExpenseForm
    onSubmit={[Function]}
  />
</div>
`;
